### PR TITLE
update documentation

### PR DESCRIPTION
**Pull request by iida-hayato**
- added what I needed about DataSource

### CODE REVIEW

File: vuepress/docs/docs/optimization.md [hunk 1]
```diff
@@ -168,3 +168,41 @@ Queries( (field) => {
...
 })
 ``` 
+
+
+### Using Optional value
+
+To use optional value with DataSource, append `.optional` to `ZQuery`
+
+```scala
+getUserNameById(id).optinal
```
Comment (ghostdogpr) [inline vuepress/docs/docs/optimization.md]:
```suggestion
getUserNameById(id).optional
```

**Review by ghostdogpr — COMMENTED** [review]
Can you target the `series/2.x` branch? It's the new default.

**Comment by iida-hayato** [comment]
@ghostdogpr thx. applied changes and change base to 2.x

**Comment by ghostdogpr** [comment]
Thanks!

**Pull request merged**